client/src/components/OptimizedImage.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
```

Comment:
So the image is generated on the fly /on demand? No pre-work? right? How about the performance when show the image lists.

Reply:
Yes, images are generated on the fly and there is no upfront work required. We probably won't see much performance increase the first time an image is loaded, but the user will benefit from smaller file sizes. Each subsequent request should be much faster since the image will be cached. 

You can see the results below. Without optimization the total images size on the list page was 698kB and load time was 4.64s, with optimization total image size went down to 363kB and load time 3.60s.

<img width="1434" alt="Screen Shot 2021-02-18 at 9 59 21 AM" src="https://user-images.githubusercontent.com/22219519/108455222-136cf500-722b-11eb-876b-7d996ae22486.png">
<img width="1430" alt="Screen Shot 2021-02-18 at 9 59 01 AM" src="https://user-images.githubusercontent.com/22219519/108455227-1536b880-722b-11eb-8ccc-cb1e814295cf.png">